### PR TITLE
feat(compute): add create template samples

### PR DESCRIPTION
**Pull request by FrodoTheTrue**

### CODE REVIEW

File: compute/instance-templates/create-instance-templates/create_template.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	"context"
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+// createInstance creates a new instance template with the provided name and a specific instance configuration.
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/create_template.go:28]: Isn't it the createTemplate function?
createTemplate creates a new instance template with....

File: compute/instance-templates/create-instance-templates/create_template.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+						},
+					},
+				},
+			},
+		},
+	}
+
+	op, err := instanceTemplatesClient.Insert(ctx, req)
+	if err != nil {
+		return fmt.Errorf("unable to create instance template: %v", err)
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/create_template.go:79]: Unable to create...
Please use sentence case.

File: compute/instance-templates/create-instance-templates/create_template.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	defer globalOperationsClient.Close()
+
+	for {
+		waitReq := &computepb.WaitGlobalOperationRequest{
+			Operation: op.Proto().GetName(),
+			Project:   projectID,
+		}
+		zoneOp, err := globalOperationsClient.Wait(ctx, waitReq)
+		if err != nil {
+			return fmt.Errorf("unable to wait for the operation: %v", err)
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/create_template.go:95]: Unable

File: compute/instance-templates/create-instance-templates/create_template_from_instance.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+	req := &computepb.InsertInstanceTemplateRequest{
+		Project: projectID,
+		InstanceTemplateResource: &computepb.InstanceTemplate{
+			Name:           proto.String(templateName),
+			SourceInstance: proto.String(instance),
+			SourceInstanceParams: &computepb.SourceInstanceParams{
+				DiskConfigs: []*computepb.DiskInstantiationConfig{
+					{
+						// Device name must match the name of a disk attached to the instance you are
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/create_template_from_instance.go:50]: How about changing this to,
Device name must match the name of a disk attached to the instance your template is based on.

File: compute/instance-templates/create-instance-templates/create_template_from_instance.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	defer globalOperationsClient.Close()
+
+	for {
+		waitReq := &computepb.WaitGlobalOperationRequest{
+			Operation: op.Proto().GetName(),
+			Project:   projectID,
+		}
+		zoneOp, err := globalOperationsClient.Wait(ctx, waitReq)
+		if err != nil {
+			return fmt.Errorf("unable to wait for the operation: %v", err)
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/create_template_from_instance.go:82]: Unable

File: compute/instance-templates/create-instance-templates/create_template_with_subnet.go [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+	defer globalOperationsClient.Close()
+
+	for {
+		waitReq := &computepb.WaitGlobalOperationRequest{
+			Operation: op.Proto().GetName(),
+			Project:   projectID,
+		}
+		zoneOp, err := globalOperationsClient.Wait(ctx, waitReq)
+		if err != nil {
+			return fmt.Errorf("unable to wait for the operation: %v", err)
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/create_template_with_subnet.go:89]: Unable

File: compute/instance-templates/create-instance-templates/delete_instance_template.go [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START compute_template_get]
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/delete_instance_template.go:17]: Please check the region tag here. Is this the sample for deleting instance template.

File: compute/instance-templates/create-instance-templates/delete_instance_template.go [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+import (
+	"context"
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+)
+
+// getInstanceTemplate retrieves an instance template, which you can use to create virtual machine
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/delete_instance_template.go:27]: I think the comments in this file should be for deleting the template.

File: compute/instance-templates/create-instance-templates/delete_instance_template.go [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+	defer globalOperationsClient.Close()
+
+	for {
+		waitReq := &computepb.WaitGlobalOperationRequest{
+			Operation: op.Proto().GetName(),
+			Project:   projectID,
+		}
+		zoneOp, err := globalOperationsClient.Wait(ctx, waitReq)
+		if err != nil {
+			return fmt.Errorf("unable to wait for the operation: %v", err)
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/delete_instance_template.go:63]: Unable

File: compute/instance-templates/create-instance-templates/delete_instance_template.go [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+		if *zoneOp.Status.Enum() == computepb.Operation_DONE {
+			fmt.Fprintf(w, "Instance template deleted\n")
+			break
+		}
+	}
+
+	return nil
+}
+
+// [END compute_template_get]
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/delete_instance_template.go:75]: Same comment as above. Please check and update. Shouldn't this be compute_template_delete?

File: compute/instance-templates/create-instance-templates/get_instance_template.go [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START compute_template_delete]
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/get_instance_template.go:17]: compute_template_get. Please check and correct the region tag for this sample.

File: compute/instance-templates/create-instance-templates/get_instance_template.go [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+import (
+	"context"
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+)
+
+// deleteInstanceTemplate deletes an instance template.
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/get_instance_template.go:27]: Use the comment for getInstanceTemplate.

File: compute/instance-templates/create-instance-templates/get_instance_template.go [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+
+	req := &computepb.GetInstanceTemplateRequest{
+		Project:          projectID,
+		InstanceTemplate: templateName,
+	}
+
+	return instanceTemplatesClient.Get(ctx, req)
+}
+
+// [END compute_template_delete]
```
Comment (savijatv) [inline compute/instance-templates/create-instance-templates/get_instance_template.go:47]: compute_template_get

File: compute/instance-templates/create-instance-templates/create_template.go (hunk 2 above)
Comment (codyoss) [reply compute/instance-templates/create-instance-templates/create_template.go:79]: Actually errors in Go should start with lower case and not use punctuation: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: compute/instance-templates/create-instance-templates/create_template.go (hunk 3 above)
Comment (codyoss) [reply compute/instance-templates/create-instance-templates/create_template.go:95]: See comment above.

File: compute/instance-templates/create-instance-templates/create_template_from_instance.go (hunk 5 above)
Comment (codyoss) [reply compute/instance-templates/create-instance-templates/create_template_from_instance.go:82]: See comment above.

File: compute/instance-templates/create-instance-templates/create_template_with_subnet.go (hunk 6 above)
Comment (codyoss) [reply compute/instance-templates/create-instance-templates/create_template_with_subnet.go:89]: See comment above.

File: compute/instance-templates/create-instance-templates/delete_instance_template.go (hunk 9 above)
Comment (codyoss) [reply compute/instance-templates/create-instance-templates/delete_instance_template.go:63]: See comment above.

**Comment by FrodoTheTrue** [comment]
@savijatv Thanks for the review! I have fixed the comments.

File: compute/instance-templates/create-instance-templates/create_instance_templates_test.go [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+	machineType := "n1-standard-1"
+	sourceImage := "projects/debian-cloud/global/images/family/debian-10"
+	networkName := "global/networks/default"
+	subnetworkName := "regions/asia-east1/subnetworks/default"
+
+	ctx := context.Background()
+
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		t.Errorf("NewInstancesRESTClient: %v", err)
```
Comment (codyoss) [inline compute/instance-templates/create-instance-templates/create_instance_templates_test.go:50]: nit: I think many(or all?) of these should be t.Fatalf. This test in others seem sequential so it does not make sense to keep running the test if you can't create the client for instance.

File: compute/instance-templates/create-instance-templates/get_instance_template.go [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+)
+
+// getInstanceTemplate retrieves an instance template, which you can use to create virtual machine
+// (VM) instances and managed instance groups (MIGs).
+func getInstanceTemplate(w io.Writer, projectID, templateName string) (*computepb.InstanceTemplate, error) {
```
Comment (codyoss) [inline compute/instance-templates/create-instance-templates/get_instance_template.go:29]: The writer here does not seems to be needed, it can be removed.

**Review by codyoss — CHANGES_REQUESTED** [review]
Just a couple of small things.

File: compute/instance-templates/create-instance-templates/create_instance_templates_test.go (hunk 14 above)
Comment (FrodoTheTrue) [reply compute/instance-templates/create-instance-templates/create_instance_templates_test.go]: For clients creating agreed.

**Review by codyoss — APPROVED** [review]

**Pull request merged**